snapcraft/parts/plugins/ubuntu_kernel_plugin.py

```diff
@@ -0,0 +1,413 @@
+            self.options.ubuntu_kernel_image_target
+            if self.options.ubuntu_kernel_image_target is not None
+            else _DEFAULT_KERNEL_IMAGE_TARGET[part_info.arch_build_for]
+        )
+
+    @override
+    def get_build_snaps(self) -> set[str]:
+        return set()
+
+    @override
+    def get_build_packages(self) -> set[str]:
+        emit.message("Getting build packages")
+        # TODO(stewarthore) extract the build dependency list from the debian
+        #  source package.
+        # https://warthogs.atlassian.net/browse/KE-427
+        build_packages = {
+            "debhelper-compat",
+            "cpio",
+            "kmod",
+            "makedumpfile",
+            "libcap-dev",
+            "libelf-dev",
+            "libnewt-dev",
+            "libiberty-dev",
+            "default-jdk-headless",
+            "java-common",
+            "rsync",
+            "libdw-dev",
+            "libpci-dev",
+            "pkg-config",
+            "python3-dev",
+            "flex",
+            "bison",
+            "libunwind8-dev",
+            "liblzma-dev",
+            "openssl",
+            "libssl-dev",
+            "libaudit-dev",
+            "bc",
+            "gawk",
+            "git",
+            "libudev-dev",
+            "autoconf",
+            "automake",
+            "libtool",
+            "uuid-dev",
+            "libnuma-dev",
+            "dkms",
+            "curl",
+            "zstd",
+            "pahole",  # not in control file
+            "bzip2",
+            "debhelper",
+            "fakeroot",
+            "lz4",
+            "python3",
+            "dwarfdump",
+        }
+
+        if self._part_info.base == "core24":
+            emit.message("Adding core24 build packages")
+            build_packages |= {
+                "python3-setuptools",
+                "libtraceevent-dev",
+                "libtracefs-dev",
+                "clang-18",
+                "rustc",
+                "rust-src",
+                "rustfmt",
+                "bindgen-0.65",
+                "libstdc++-13-dev",
+            }
+
+        if self._part_info.is_cross_compiling:
+            emit.message("Adding cross-compilation build packages")
+            build_packages |= {
+                f"binutils-{self._part_info.arch_triplet_build_for}",
+                f"gcc-{self._part_info.arch_triplet_build_for}",
+                f"libc6-dev-{self._part_info.target_arch}-cross",
+            }
+            if self._part_info.base == "core24":
+                build_packages |= {
+                    f"libstdc++-13-dev-{self._part_info.target_arch}-cross",
+                }
+
+        return build_packages
+
+    @override
+    def get_build_environment(self) -> dict[str, str]:
+        """Returns additional build environment variables."""
+        emit.message("Getting build environment")
+        return (
+            {}
+            if not self._part_info.is_cross_compiling
```

Comment:
What's the effect of including these variables even if we're not cross-compiling?